samples: net: cloud: aws_iot_mqtt: use auto init

maass-hamburg opened the pull request:
use auto init for networking, dhcp and sntp.

maass-hamburg committed:
samples: net: cloud: aws_iot_mqtt: use auto init

use auto init for networking, dhcp and sntp.
This simplifys the code of this sample by using the
auto init feature of the networking stack.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>

jukkar commented:
Could you add in the commit message why this is needed, it is far from obvious atm?

maass-hamburg committed:
samples: net: cloud: aws_iot_mqtt: remove redundant kconfig

CONFIG_ENTROPY_DEVICE_RANDOM_GENERATOR is on
by default, if the device supports that, so there is no need
to enable it explicit.

Signed-off-by: Fin Maaß <f.maass@vogl-electronic.com>
